plutus-tx/src/Language/PlutusTx/Ratio.hs

```diff
@@ -182,23 +246,24 @@ truncate r = let (m, _ :: Rational) = properFraction r in m
 properFraction :: Ratio Integer -> (Integer, Ratio Integer)
 properFraction (n :% d) = (q, r :% d) where (q, r) = quotRem n d
 
-{-# INLINABLE quot #-}
-quot :: Integer -> Integer -> Integer
-quot x y = let (q, _ :: Integer) = quotRem x y in q
+{-# INLINABLE divMod #-}
+-- | simultaneous div and mod
+-- TODO. This is doing twice as much work as it needs to: the Plutus Core
+-- builtins 'divideInteger' and 'modInteger' are implemented as the Haskell
+-- 'div' and 'mod' operations, which in turn are impemented as fst . divMod and
+-- snd . divMod, so we're calling Haskell's 'divMod' twice and throwing away
+-- useful information.  We could fix his by exposing 'divMod' as a PLC builtin,
+-- but that would require us to implement pairs as built-in types along with
+-- built-in introduction and elimination functions.  This would be non-trivial.
+divMod :: Integer -> Integer -> (Integer, Integer)
```

Comment:
Why not put these in the prelude module?

Reply:
Because `Ratio` (following `GHC.Real`) needs `quotRem` and it was already defined in there (although it was defined in terms of `divide` and `remainder` and needed a workaround to compensate for that, which is one of the things I wanted to fix: see https://github.com/input-output-hk/plutus/blob/a1ecf0431de98cca3cf77522d94a4507dd5f131f/plutus-tx/src/Language/PlutusTx/Ratio.hs#L191).  It seemed sensible to put `divMod` in the same place (and also I see that in GHC, both `divMod` and `quotRem` are defined in `GHC.Real`).

Reply:
Yeah, but they're both perfectly good functions that work on `Integer`s, you might want them regardless of whether you're working with `Ratio` or not.

Reply:
`Ratio` can import the prelude!

Reply:
Oh, the prelude imports `Ratio`. Um... I feel like this is wrong, but I'm not inspired enough to suggest a better solution. I guess they're fine here, at least they're exported from the prelude.

Reply:
> Oh, the prelude imports `Ratio`.

Exactly!

Reply:
> Um... I feel like this is wrong, but I'm not inspired enough to suggest a better solution. 

We could have one copy of `quotRem` in `Prelude` and another in `Ratio`, but let's not.